samples/Meeting/src/app/App.tsx

```diff
@@ -0,0 +1,187 @@
+        setCallUserId(user);
+      } catch (e) {
+        console.error(e);
+        setUserCredentialFetchError(true);
+      }
+    })();
+  }, []);
+
+  const ERROR_TEXT_THREAD_NOT_RECORDED = 'Thread id is not recorded in server';
+  const ALERT_TEXT_TRY_AGAIN = "You can't be added at this moment. Please wait at least 60 seconds to try again.";
+
+  const onCreateThread = async (): Promise<string> => {
+    const exisitedThreadId = await getThreadId();
+    if (exisitedThreadId && exisitedThreadId.length > 0) {
+      if (!(await checkThreadValid(exisitedThreadId))) {
+        throw new Error(ERROR_TEXT_THREAD_NOT_RECORDED);
+      }
+      return exisitedThreadId;
+    }
+
+    const threadId = await createThread();
+    if (!threadId) {
+      console.error('Failed to create a thread, returned threadId is undefined or empty string');
+      return '';
+    } else {
+      return threadId;
+    }
+  };
+
+  useEffect(() => {
+    const internalSetupAndJoinChatThread = async (): Promise<void> => {
+      if (displayName && userId) {
+        const newThreadId = await onCreateThread(); // change name of onCreateThread
+        const result = await joinThread(newThreadId, userId.communicationUserId, displayName);
+        if (!result) {
+          alert(ALERT_TEXT_TRY_AGAIN);
+          return;
+        }
+        setEndpointUrl(await getEndpointUrl());
+        setThreadId(newThreadId);
+        window.history.pushState({}, document.title, window.location + `&threadId=${newThreadId}`);
+      }
+    };
+    internalSetupAndJoinChatThread();
+  }, [displayName, userId]);
+
+  const supportedBrowser = !isOnIphoneAndNotSafari();
+  if (!supportedBrowser) {
+    return <UnsupportedBrowserPage />;
+  }
+
+  if (isMobileSession() || isSmallScreen()) {
+    console.log('ACS Meeting sample: This is experimental behaviour');
+  }
+
+  switch (page) {
+    case 'home': {
+      document.title = `home - ${webAppTitle}`;
+      // Show a simplified join home screen if joining an existing call
+      const joiningExistingCall: boolean = !!getGroupIdFromUrl() || !!getTeamsLinkFromUrl();
+      return (
+        <HomeScreen
+          joiningExistingCall={joiningExistingCall}
+          startMeetingHandler={(callDetails) => {
+            setDisplayName(callDetails.displayName);
+            const isTeamsCall = !!callDetails.teamsLink;
+            const callLocator =
+              callDetails.teamsLink || getTeamsLinkFromUrl() || getGroupIdFromUrl() || createGroupId();
+            setCallLocator(callLocator);
+
+            // Update window URL to have a joinable link
+
+            if (!joiningExistingCall) {
+              const joinParam = isTeamsCall
+                ? '?teamsLink=' + encodeURIComponent((callLocator as TeamsMeetingLinkLocator).meetingLink)
+                : '?groupId=' + (callLocator as GroupCallLocator).groupId;
+
+              window.history.pushState({}, document.title, window.location.origin + joinParam);
```

Comment:
we now have both a group id and a threadID being added to the url. Currently because this `joinParam` is prefixed with `?` it _must_ be added first and `threadId` has `&` which _means_ it must be added after.

To avoid having this unecessary restriction and prevent issues if other Query Params want to be included in the future consider using the [URL search params API](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)

So this would look something like:

```typescript
const joinParam = isTeamsCall ? 'teamsLink' : 'groupId';
const joinValue = isTeamsCall ? encodeURIComponent((callLocator as TeamsMeetingLinkLocator).meetingLink) : (callLocator as GroupCallLocator).groupId;
pushQSPToUrl({name: joinParam, value: joinValue}); // see definition below

// (As a util helper function defined outside of this scope block)
/**
 * Append a new query parameter to the existing url of the document.
 * remarks this uses window.location.pushState and not replaceState
 */
const pushQSPToUrl = (qsp: {name: joinParam, value: joinValue}) => {
  const url = new URL(window.location);
  url.searchParams.append(qsp);
  window.history.pushState({}, document.title, qsp);
};
```